src/cwrapper.h

```diff
@@ -7,15 +7,34 @@
 extern "C" {
 #endif
 
-typedef struct CWrapper CWrapper;
+// The size of 'basic_struct' must be the same as RCP<const Basic> *and* they
+// must have the same alignment (because we allocate RCP<const Basic> into the
+// memory occupied by this struct in cwrapper.cpp). We cannot use C++ in this
+// file, so we need to use C tools to arrive at the correct size and alignment.
+// The size of the RCP object on most platforms should be just the size of the
+// 'T *ptr_' pointer that it contains (as there is no virtual function table)
+// and the alignment should also be of a pointer. So we just put 'void *data'
+// as the only member of the struct, that should have the correct size and
+// alignment.  However, this is checked at compile time in cwrapper.cpp, so if
+// the size or alignment is different on some platform, the compilation will
+// fail --- in that case one needs to modify the contents of this struct to
+// adjust its size and/or alignment.
+typedef struct
+{
+    void *data;
+} basic_struct;
```

Comment:
I think we still do, as we need to create an array using it, so that the user can allocate that on the stack.